src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(27);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+    /*
+    Testing the login field
+     */
+
+    @Test
+    void register_loginDetailsAreCorrect_Ok() {
+        user.setLogin(DEFAULT_LOGIN);
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+        User expected = storageDao.get(DEFAULT_LOGIN);
+        assertEquals(expected, actual);
+
+    }
+
+    @Test
+    void register_user_notOk() {
+        Storage.people.add(user);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_withNull_notOk() {
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    void register_emptyLineLogin_notOk() {
+        user.setLogin("  ");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_checkForAbsenceInTheFirstLetterOfCharacter_notOk() {
+        user.setLogin("#oleksii");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+    /*
+    Testing the password field
+     */
+
+    @Test
+    void register_withCorrectPassword_ok() {
+        user.setPassword(DEFAULT_PASSWORD);
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void register_withNullInPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_withEmptyFiled_notOk() {
+        user.setPassword("  ");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_withVeryLongPassword_notOk() {
+        user.setPassword("1234567891011121314");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_withVerySmallPassword_notOk() {
+        user.setPassword("12");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+    /*
+    Testing the age field
+     */
+
+    @Test
+    void register_underageUser_notOk() {
+        user.setAge(12);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_veryOldUser_notOk() {
+        user.setAge(90);
```

Comment:
What is the max acceptable age according to the task?

Reply:
the task does not specify the maximum age, I myself decided to add this test, if we follow strictly according to the task, then I delete it